src/repository/resources/ipres.rs

```diff
@@ -1614,6 +1614,178 @@ impl AddressFamily {
 }
 
 
+//------------ Ipv4Blocks ----------------------------------------------------
+
+/// Contains IPv4 resources. This type is a thin wrapper around the underlying
+/// [`IpBlocks`] type intended to help with serializing/deserializing and
+/// formatting using IPv4 syntax.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Ipv4Blocks(IpBlocks);
```

Comment:
I wonder if we can employ the idea of being generic over a marker type here as well – `IpBlocks<T>` where T is one of `V4` or `V6`?

Reply:
I don't see a quick way to do this. We already have `IpBlocks` which is not generic. Changing that would break many things.

Unless you have concerns about this I prefer to leave it as is.